### ISSUE TITLE
CMake: Heffte include directories not passed to targets

### ISSUE THREAD

**Issue by sfogerty** [ECP-copa/Cabana#259]
When building CabanaMD and including Cabana-Cajita's`FastFourierTransform.hpp`, which itself includes `heffte_fft3d.h`, a build error occurs.

Here is how I am building Cabana:
```cmake
cmake \
     -D CMAKE_BUILD_TYPE="Debug" \
     -D CMAKE_PREFIX_PATH="$KOKKOS_INSTALL_DIR;$HEFFTE_INSTALL_DIR" \
     -D CMAKE_INSTALL_PREFIX=$CABANA_INSTALL_DIR \
     -D Cabana_ENABLE_MPI=ON \
     -D Cabana_ENABLE_TESTING=ON \
     -D Cabana_ENABLE_EXAMPLES=ON \
     -D Cabana_ENABLE_CAJITA=ON \
     -D Cabana_ENABLE_HEFFTE=ON \
     .. ;
```
 and CabanaMD: 
```cmake
cmake -D CMAKE_PREFIX_PATH="$CABANA_INSTALL_DIR" \
    -D CabanaMD_ENABLE_Serial=OFF \
    -D CabanaMD_ENABLE_OpenMP=ON \
    -D CabanaMD_ENABLE_Cuda=OFF \
    -D CabanaMD_ENABLE_LongRange=ON \
    .. ;
```

#258 is in progress and will fix this, unless the cause is actually something in CabanaMD itself or an error in how I'm building.

Error prints: `Cajita_FastFourierTransform.hpp:20:26: fatal error: heffte_fft3d.h: No such file or directory`

**Comment by sslattery** [comment]
try `-D Cabana_REQUIRE_HEFFTE=ON` in the Cabana CMake and see what that changes.

**Comment by sfogerty** [comment]
adding `-D Cabana_REQUIRE_HEFFTE=ON` had an impact, and exposed a bug in Heffte. However, CabanaMD still  cant find `heffte_fft3d.h`.

The change I made to `heffte/cmake/HeffteConfig.cmake`:
``` cmake
if(NOT TARGET Heffte::FFTW)
   add_library(Heffte::FFTW INTERFACE IMPORTED GLOBAL)
endif()
```

**Comment by junghans** [comment]
I think we need to make heffte a `PUBLIC` when linking in Cabana, so that its include directory gets forwarded to CabanaMD.

**Comment by sfogerty** [comment]
This issue has been fixed by #258.

As for the note above:
> adding `-D Cabana_REQUIRE_HEFFTE=ON` had an impact, and exposed a bug in Heffte. 

...the issue was in my build. I added `-D Cabana_REQUIRE_HEFFTE=ON` in addition to `-D Cabana_ENABLE_HEFFTE=ON`. This will cause an error when the CMake Cabana build to try to create a Heffte target library of the same name twice.

I think it's fair to expect users to build with either `Cabana_REQUIRE_HEFFTE`, or `Cabana_ENABLE_HEFFTE` and similar pairs, but not use both. Maybe we can note this in documentation.
